### ISSUE TITLE
Section about working with packages locally

### ISSUE THREAD

**Issue by splendido** [percolatestudio/atmosphere#312]
Dear all,

It happened to me at least a couple of times to see some guy forking a package and publishing it only to test their local changes for their current project.

Do you think adding one page or one section to the [Publishing Packages](https://atmospherejs.com/i/publishing) guide could help people start off quicker without needing to publish packages even before knowing if they'll actually need them?
...actually there's already something under the _Testing your package_ section but perhaps the context does not make it that much clear the same approach can be used to test _modified_ cloned/forked packages.

Thought?

I've recently added a [Working Locally](https://github.com/meteor-useraccounts/core#working-locally) section to one of my README files, basically elaborating on what published by @tmeasday and @cmather [here](https://github.com/EventedMind/iron-router#working-locally).

I think you guys could do a great job here!
Tnx

**Comment by dburles** [comment]
Hey @splendido I think this is a good idea, I could see it being its own section on that page.

**Comment by dburles** [comment]
@tmeasday is there any consensus on a reasonably straightforward approach to working with packages locally?

**Comment by tmeasday** [comment]
I think `git clone`-ing them into your project's `packages/` directory probably makes sense if you are tinkering. If you end up needing to check it in, a submodule works, but I guess other people have different approaches.

**Comment by splendido** [comment]
Yeah anyone could have his preferred way of doing it.

What about simply explaining how Meteor could find your local version of the package?

Basically finding something under the `packages` folder of your project (be it a git clone, git submodule, proper folder under development, a symbolic link, whatever...)
...or looking into the path provided via `PACKAGE_DIRS` env variable

**Comment by dburles** [comment]
Yeah sounds good to me. Also some procedures may help those that might otherwise be quick to publish a fork.

**Comment by splendido** [comment]
Yep, that was it!

Taking people by hands about how to test changes to forked packages before they publish them only for tinkering... ;-)

**Comment by dandv** [comment]
Let's pick one method, that also works on Windows easily (since Windows will probably see more Meteor noobs). Even though Windows supports symlinks, they're not really popular, so I'd recommend teaching users to 
1. `mkdir packages`
2. `cd packages`
3. `git clone <repo_url>`

**Comment by mitar** [comment]
Should the 3. be:

```
git clone <repo_url> <package_prefix>:<package_name>
```

So the directory should match the package name, and not be repository name default? I think I had issues with this in the past.

**Comment by splendido** [comment]
So far it worked fine also with different names I used, but I guess there might be some edge cases...

**Comment by mitar** [comment]
It works if the package has `name` defined in the `package.js`. If it does not, then the directory name has to match the full package name. Otherwise `meteor` silently picks up the official package, not your fork in the `packages`. This is why I am recommending that for novices we say that they have to clone with a package name as a directory name. Then it will work for them for all packages.

BTW, why the above commands would not be:

```
git init
git submodule add <repo_url> packages/<package_prefix>:<package_name>
```

You do not have to create manually `packages` directory then. And it is much better than cloning (because cloning inside an existing `git` repository might get things messy). You can also commit it then and share with the world. (Good for reproductions.) (It is true that committing inside a submodule might be tricky for some.)

**Comment by splendido** [comment]
...and you found them! ;-)

**Comment by tmeasday** [comment]
I think @mitar is right about the name, just in case.

I'm not sure about the submodule. My feeling is that the cons outweigh the pros (at lot of new developers may never have used a submodule and not even understand the state their repo is now in). Then again the cons of cloning inside a git repo are kind of bad too. Not sure.
